File: backend/models/Song.cs

namespace backend.models
{
	public class Song
	{
		public required string FileName { get; set; }
		public string? Title { get; set; }
		public DateTime? CreationDate { get; set; }
		public string? Album { get; set; }
		public int Rating { get; set; }
		public string? Artist { get; set; }
	}
}

File: Frontend/src/app/models/song.ts
import { SafeUrl } from "@angular/platform-browser";

export interface Song {
  fileName: string;
  title?: string;
  creationDate: Date;
  album?: string;
  isLiked: boolean;
  rating: number;
  artist?: string;
  duration: number;
  imageData?: Uint8Array;
  imageUrl?: SafeUrl;
}
